lazy_tensor_core/lazy_tensor_core/csrc/LazyShapeDtype.cpp

```diff
@@ -51,6 +51,18 @@ namespace torch_lazy_tensors {
 namespace ir {
 namespace ops {
 
+std::vector<std::vector<int64_t>> compute_shape_masked_fill(at::Tensor & self, const at::Tensor & mask, const at::Scalar & value) {
+  return {self.sizes().vec()};
+}
+std::vector<c10::ScalarType> compute_dtype_masked_fill(at::Tensor & self, const at::Tensor & mask, const at::Scalar & value) {
+  return {value.type()};
```

Comment:
Could the dtype of self different than the dtype of value?

Reply:
I guess so. let me double check my assumption. i assumed that value's dtype would overwrite whatever self's is.

Reply:
oops, good catch. i was way off. It is always self.type, and if new value does not fit, it throws an error